### ISSUE TITLE
Error running job: Send daily survey. Due to: (_mysql_exceptions.OperationalError) (2006, 'MySQL server has gone away')

### ISSUE THREAD

**Issue by PauloPhagula** [ntxuva/mopa-utils#7]
```
Error running job: Send daily survey.
Error message:(_mysql_exceptions.OperationalError) (2006, 'MySQL server has gone away') [SQL: u'SELECT mopa_survey.id AS mopa_survey_id, mopa_survey.created_at AS mopa_survey_created_at, mopa_survey.updated_at AS mopa_survey_updated_at, mopa_survey.survey_id AS mopa_survey_survey_id, mopa_survey.survey_type AS mopa_survey_survey_type, mopa_survey.district AS mopa_survey_district, mopa_survey.neighbourhood AS mopa_survey_neighbourhood, mopa_survey.point AS mopa_survey_point, mopa_survey.question AS mopa_survey_question \nFROM mopa_survey ORDER BY mopa_survey.id DESC \n LIMIT %s'] [parameters: (1,)].
Exception Type: <class 'sqlalchemy.exc.OperationalError'>.
File name: tasks.py.
Line No: 33.
Traceback: Traceback (most recent call last):
  File "/home/psms1/sms-app/mopa/tasks.py", line 33, in wrapper
    return func(*args, **kwargs)
  File "/home/psms1/sms-app/mopa/tasks.py", line 452, in send_daily_survey
    survey = Survey(survey_type="G", question=constants.SMS_INTRO)
  File "<string>", line 4, in __init__
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/orm/state.py", line 306, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/orm/state.py", line 303, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/home/psms1/sms-app/mopa/models.py", line 148, in __init__
    self.survey_id = self.get_next_survey_id()
  File "/home/psms1/sms-app/mopa/models.py", line 164, in get_next_survey_id
    last_survey = self.get_last_survey()
  File "/home/psms1/sms-app/mopa/models.py", line 172, in get_last_survey
    return self.query.order_by(desc(Survey.id)).first()
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2445, in first
    ret = list(self[0:1])
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2281, in __getitem__
    return list(res)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2516, in __iter__
    return self._execute_and_instances(context)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2531, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
    exc_info
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/home/psms1/.local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
  File "/home/psms1/.local/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/psms1/.local/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
OperationalError: (_mysql_exceptions.OperationalError) (2006, 'MySQL server has gone away') [SQL: u'SELECT mopa_survey.id AS mopa_survey_id, mopa_survey.created_at AS mopa_survey_created_at, mopa_survey.updated_at AS mopa_survey_updated_at, mopa_survey.survey_id AS mopa_survey_survey_id, mopa_survey.survey_type AS mopa_survey_survey_type, mopa_survey.district AS mopa_survey_district, mopa_survey.neighbourhood AS mopa_survey_neighbourhood, mopa_survey.point AS mopa_survey_point, mopa_survey.question AS mopa_survey_question \nFROM mopa_survey ORDER BY mopa_survey.id DESC \n LIMIT %s'] [parameters: (1,)]
```

**Comment by PauloPhagula** [comment]
According to documentation from https://dev.mysql.com/doc/refman/5.7/en/gone-away.html and other threads this has two main causes
- Server timed out and closed the connection. Time out is set with “wait_timeout” mysql variable in my.cnf so maybe it's not large enough.
- Server dropped an incorrect or too large packet. If mysqld gets a packet that is too large or incorrect, it assumes that something has gone wrong with the client and closes the connection. The maximal packet size limit is set with “max_allowed_packet” variable in my.cnf.

As such `wait_timeout` and `mat_allowed_packet` where increased and MySQL was restarted.

But more over in our particular case the query went over the `wait_timeout` because there was a transaction locking the table, thus stopping other queries from running. Said transaction is the same that caused issue #8 .

So basically once the first job failed and caused the lock, All jobs that came after it failed too.
